src/gui/topbar_subwindows/gui_diplomacy_window.hpp

```diff
@@ -1452,6 +1452,15 @@ class war_side_strength_text : public button_element_base {
 	void update_tooltip(sys::state& state, int32_t x, int32_t y, text::columnar_layout& contents) noexcept override {
 		dcon::war_id content = retrieve<dcon::war_id>(state, parent);
 
+		sys::year_month_day start_date = o.get_war().get_start_date().to_ymd();
```

Comment:
you can just write the base text with a single variable to substitute and then pass the date in directly to the substitution map without converting it into anything. It will then be expanded to a formatted date for the variable (just like we do for country ids -> names)